### PR TITLE
Support to cache method LocalCacheView.toCacheKey(Object) for CacheProvider.REDISSON

### PR DESCRIPTION
**Pull request by lehuuthanh5**
This will help to increase the performance of RedissonLocalCachedMap by introducing a cache synonym on `LocalCacheView.toCacheKey(Object)`  method.

Changes:

- Add new config `LocalCachedMapOptions.storeCacheKey` that defines whether to store CacheKey of an object key into the local cache.
- Create a new local cache for caching CacheKey purpose
- Inside `LocalCacheView.toCacheKey(Object)`:
    - Check if the Object already has Its CacheKey, and return the cached value.
    - If the Object does not have its CacheKey, create a new CacheKey put it into the local cache, and return it.

### CODE REVIEW

File: redisson/src/main/java/org/redisson/api/LocalCachedMapOptions.java [hunk 1]
```diff
@@ -400,6 +407,18 @@ public LocalCachedMapOptions<K, V> storeCacheMiss(boolean storeCacheMiss) {
...
     }
 
+    /**
+     * Defines whether to store CacheKey of an object key into the local cache. <br>
+     * This indicator only affects when {@link LocalCachedMapOptions#cacheProvider} != {@link CacheProvider#CAFFEINE}
+     *
+     * @param storeCacheKey - whether to store CacheKey of an object key into the local cache
+     * @return LocalCachedMapOptions instance
+     */
+    public LocalCachedMapOptions<K, V> storeCacheKey(boolean storeCacheKey) {
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/api/LocalCachedMapOptions.java:417]: Should be renamed to `useObjectAsKey()`

**Review by mrniko — CHANGES_REQUESTED** [review]

File: redisson/src/main/java/org/redisson/RedissonLocalCachedMap.java [hunk 2]
```diff
@@ -80,12 +80,12 @@ public RedissonLocalCachedMap(Codec codec, CommandAsyncExecutor connectionManage
 
     private void init(LocalCachedMapOptions<K, V> options, EvictionScheduler evictionScheduler) {
         if (options.getCacheProvider() == LocalCachedMapOptions.CacheProvider.CAFFEINE) {
-            options.storeCacheKey(false);
+            options.useObjectAsCacheKey(false);
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonLocalCachedMap.java:83]: IllegalArgumentException should be thrown if this option can't be used with specific cache provider.

**Review by mrniko — CHANGES_REQUESTED** [review]

File: redisson/src/main/java/org/redisson/RedissonLocalCachedMap.java (hunk 2 above)
Comment (lehuuthanh5) [reply redisson/src/main/java/org/redisson/RedissonLocalCachedMap.java]: Done

**Review by mrniko — APPROVED** [review]

**Comment by mrniko** [comment]
Thanks for contribution!

**Pull request merged**